### ISSUE TITLE
Container not starting with error "Missing X server or $DISPLAY"

### ISSUE THREAD

**Issue by Schnabulation** [guysoft/MagicMirrorOS#67]
Brand new installation of MagicMirror OS on a brand new Raspberry Pi 5 (4GB) and a brand new SanDisk 32GB SD card. System has internet, I can SSH into it. It does pull the images but upon starting the docker container "mm" it crashes with the error message `Missing X server or $DISPLAY`

I have then checked the container "xserver", which crashes with `(EE) no screens found(EE)`.

On the screen (normal PC screen, connected to HDMI0 via a Micro-HDMI to HDMI cable) I can see the startup screen of linux and also get the prompt to login. However Xserver never starts.

Any idea?

**Comment by khassel** [comment]
> Any idea?

not really, main problem is I have no raspi5 to test.

Downloaded the latest image `2024-07-04-magicmirroros-bookworm-arm64-lite-1.0.0.zip | 843.85 MiB | Sep 28, 2024, 6:17 AM`, put it on sd card and it works out of the box ...

Did you use the `arm64` or `armhf` image?

Goggling the error brings many results ...

**Comment by khassel** [comment]
I added an additional config file to the xserver container, so you could try:

```bash
cd /opt/mm/run
docker compose pull
docker compose up -d
```

**Comment by Schnabulation** [comment]
I have a RPi4 and RPi5, I can test both - but will take some time.

Thank you for your help!

**Comment by guysoft** [comment]
@khassel need a new build?

**Comment by khassel** [comment]
> @khassel need a new build?

no, we changed the build link in the readme to gitlab so they are under my control ...

This problem is related to pi5 and maybe the xserver container

**Comment by guysoft** [comment]
@khassel cool perhaps you want to update the paths in https://github.com/guysoft/pi-imager-web/pull/2
This would let people flash the image using pi-imager unofficial imager. But it means you need to generate the json for that.

**Comment by khassel** [comment]
xserver was replaced by labwc, ping me for reopening if the issue persists.
